src/main/java/seedu/duke/operationlist/PassengerList.java

```diff
@@ -0,0 +1,437 @@
+package seedu.duke.operationlist;
+
+import seedu.duke.exceptions.SkyControlException;
+import seedu.duke.terminalinfo.PassengerInfo;
+import seedu.duke.ui.Ui;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class PassengerList extends OperationList {
+    protected static final String REGEX_LETTER = "[a-zA-Z ]*";
+    protected static final String REGEX_NUMBER = "[0-9]*";
+    protected static final String REGEX_TIME = "([01]?[0-9]|2[0-3])[0-5][0-9]";
+    protected static final String NAME_DELIMITER = "n/";
+    protected static final String DEPARTURE_TIME_DELIMITER = " dt/";
+    protected static final String FLIGHT_NUMBER_DELIMITER = " fn/";
+    protected static final String GATE_NUMBER_DELIMITER = " gn/";
+    protected static final String BOARDING_GROUP_DELIMITER = " bg/";
+    protected static final String SEAT_NUMBER_DELIMITER = " sn/";
+    protected static final String BOARDING_TIME_DELIMITER = " bt/";
+    protected static final String EMPTY_STRING = "";
+    public static final int NO_PASSENGER = 0;
+    protected static final int NAME_LENGTH_LIMIT = 24;
+    protected static final int FN_MIN_LENGTH = 4;
+    public static final int FIRST_INDEX = 0;
+    public static final int SECOND_INDEX = 1;
+    public static final int MIN_FN_LENGTH = 2;
+    public static final int MAX_FN_LENGTH = 4;
+    public static final int MAX_GN_LENGTH = 2;
+    public static final int MIN_GN_LENGTH = 1;
+    public static final int SEAT_NUM_SECOND_INDEX = 1;
+    public static final int SEAT_NUM_FIRST_INDEX = 0;
+    public static final int SEAT_NUM_THIRD_INDEX = 2;
+    protected static boolean isNamePresent = false;
+    protected static boolean isFlightNumberPresent = false;
+    protected static boolean isSeatNumberPresent = false;
+    protected static boolean isDepartureTimePresent = false;
+    protected static boolean isValidPassenger = false;
+    protected static boolean isDeleteSuccess = false;
+    protected static boolean isExceedNameLength = false;
+    protected static boolean isWrongNameFormat = false;
+    protected static boolean isWrongDepartureFormat = false;
+    protected static boolean isWrongFlightNumFormat = false;
+    protected static boolean isWrongBoardingTimeFormat = false;
+    protected static boolean isWrongGateNumberFormat = false;
+    protected static boolean isWrongBoardingGroupFormat = false;
+    protected static boolean isWrongSeatNumberFormat = false;
+    protected String name;
+    protected String departureDate;
+    protected String departureTime;
+    protected static String flightNumber;
+    protected String gateNumber;
+    protected String seatNumber;
+    protected String boardingTime;
+    protected int numOfPassengers = 0;
+    protected int boardingGroup;
+    protected static int startIndex;
```

Comment:
Consider minimising the scope of variables. For instance, declaring a boolean variable (not constants) only when it will be used within the function. If the variables are required, you can consider removing some or saving them into an object that parses into a function when needed to lower the global variable counts. 

> Minimize global variables. Global variables may be the most convenient way to pass information around, but they do create implicit links between code segments that use the global variable. Avoid them as much as possible.